### PR TITLE
[Matrix] Convert column-vector ops feeding dot product to row-vectors.

### PR DESCRIPTION
**Pull request by fhahn**
Generalize the logic used to convert column-vector ops to row-vectors to support converting chains of operations.

A potential next step is to further generalize this to convert column-vector ops to row-vector ops in general, not just for operands of dot products. Dot-product handling would then be driven by the general conversion, rather than the other way around.

### CODE REVIEW

**Comment by fhahn** [comment]
ping :)

**Comment by fhahn** [comment]
ping :)

File: llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp [hunk 1]
```diff
@@ -1391,7 +1394,26 @@ class LowerMatrixIntrinsics {
       return TTI.getMemoryOpCost(Instruction::Load, VecTy, Align(1), 0) -
              N * TTI.getMemoryOpCost(Instruction::Load, EltTy, Align(1), 0);
     };
-    auto LHSCost = GetCostForArg(LHS, LShape.NumColumns);
+
```
[the remark this answers is not in the document]
Comment (fhahn) [reply llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp:1397]: Thanks, added a comment for what the loop does, does hat make sense?

[the remark this answers is not in the document]
Comment (francisvm) [reply llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp:1397]: Great, I was missing the "difference" part. Thanks!

**Review by francisvm — APPROVED** [review]

**Pull request merged**